### DISCUSSION TITLE
Calculations of fields

### DISCUSSION THREAD

**Discussion by milotype** [WeblateOrg/weblate#6222] [Ideas]
**Describe the issue**

1. Some fields do not show any content/value
2. Number of contributors is **zero**!
3. Calculations mismatch

**To Reproduce the issue**

Go to 'https://hosted.weblate.org/projects/openbikesharing/#information' (maybe this issue applies to other projects as well)

**Expected behavior**

1. Fields should not be emty
2. Number of contributors should be calculated. If 92% is transalated, it is _impossible_ that there is no contributor involved. Even if the project has only the source strings (mostly in english), the minimum for this field should be at least 1.
3. If there is a percentage for "Trends ...", there should be a number/value in the above field as well.

**Screenshots**

![Calculations](https://user-images.githubusercontent.com/43657314/122460277-36799c80-cfb2-11eb-84f1-5f2b82428f11.png)

**Comment by comradekingu** [comment]
Can confirm, the strings translated was too high in Remmina.

@milotype Could it be that 92% got imported and wasn't translated in Weblate?
Or 92% is read-only strings?

"Hosted strings" and "Hosted words" for last 30 days should be populated?
Or is it only new ones?

**Comment by nijel** [comment]
1. Why show 100% there? Showing it adds no value.
2. It's contributors in last 30 days.
3. A dash is shown when there is no change in the value for 30 days.

**Comment by comradekingu** [comment]
"30 day trend" or "last 30 days" makes more sense to me.
Isn't the top field for "translated" in the last 30 supposed to have something?

**Comment by nijel** [comment]
>  Isn't the top field for "translated" in the last 30 supposed to have something?

It behaves as expected now - there was no change in last 30 days in this. Maybe dash is not the best indication of this, though...

**Comment by milotype** [comment]
>     1. Why show 100% there? Showing it adds no value.
True, but a field should IMO allways have a value ... otherwise it looks like a mistake/error to me ...

**Comment by nijel** [comment]
Any listing in Weblate also shows blank space instead of showing zeroes:

![image](https://user-images.githubusercontent.com/212189/122787186-5d3b1a00-d2b5-11eb-8834-53eb158df5ad.png)

**Comment by comradekingu** [comment]
I was thinking "changes" instead of "trends" with something like
![bilde](https://user-images.githubusercontent.com/13802408/122791399-b9e40800-d2a8-11eb-8b65-6d3c171afada.png)
or the up and down symbols with that flat thing in the middle?

![bilde](https://user-images.githubusercontent.com/13802408/122792283-8d7cbb80-d2a9-11eb-8c9d-6c45cb8672a3.png)

![bilde](https://user-images.githubusercontent.com/13802408/122792610-d7fe3800-d2a9-11eb-8c69-24cde0f734e1.png)

Possibly washed out top and down colours.

If everything but the trending direction was grayed out, and everything was in black and white there wouldn't be a red/green colourblind problem to begin with.

![bilde](https://user-images.githubusercontent.com/13802408/122793652-fc0e4900-d2aa-11eb-8cdb-e7ec7ea4fa6e.png)


In norway the -||- symbol is used for same, but that is usually repetition of an already mentioned item in a list.

Edit: I don't know what the bottom row of percentages signifies.
Ah, that is the previous numbers.

**Comment by milotype** [comment]
> Any listing in Weblate also shows blank space instead of showing zeroes:
> 
> ![image](https://user-images.githubusercontent.com/212189/122787186-5d3b1a00-d2b5-11eb-8834-53eb158df5ad.png)

But why do you want to have, or what is the use of empty fields?? Think about blind persons ... empty fields have no content!

**Comment by nijel** [comment]
Because it attracts attention to the non-empty fields, which are more important. To address accessibility, `sr-only` content could be added, but that's yet another topic mixed into this issue.

**Comment by milotype** [comment]
Yes, I aggree about your explanation about attracting attention ... Good point!

I'm sorry I can't really help with implementing `sr-only` content ... I'm no UI expert, so just as a suggestion:

- if a field is empty, screenreaders should read something like "This field is empty" or "100%" or "Zero" (depending upon the contents)
- if a field consists of a dash character, screenreaders should read something like "None" or "Zero".

**Reply by nijel** [reply]
I've added that to the listings in fc4f949520.
